arrays/arrays.c

```diff
@@ -105,11 +105,16 @@ void arr_append(Array *arr, char *element) {
 
   // Resize the array if the number of elements is over capacity
   // or throw an error if resize isn't implemented yet.
-
+  
+  if (arr->capacity <= arr->count) {
+      fprintf(stderr, "IndexError: Index is out of range");
+      return;
+  }
   // Copy the element and add it to the end of the array
+  char *new_element = strdup(element);
```

Comment:
Remember to add this element to the end of the array before increasing the count.